src/room_membership/join_room/v1.rs

```diff
@@ -0,0 +1,84 @@
+//! [POST /_synapse/admin/v1/join/:room_id_or_alias](https://github.com/element-hq/synapse/blob/master/docs/admin_api/room_membership.md)
+
+use ruma::{
+    api::{request, response, Metadata},
+    metadata, OwnedRoomId, OwnedRoomOrAliasId, OwnedUserId,
+};
+use serde::{Deserialize, Serialize};
+
+const METADATA: Metadata = metadata! {
+    method: POST,
+    rate_limited: false,
+    authentication: AccessToken,
+    history: {
+        unstable => "/_synapse/admin/v1/join/:room_id_or_alias",
+    }
+};
+
+#[request]
+#[derive(Serialize, Deserialize, PartialEq)]
```

Comment:
Request and response types should not implement `Serialize`, `Deserialize`. The `#[request]` and `#[response]` macros will implement the relevant traits for HTTP (de)serialization.

If you want to test the HTTP (de)serialization, see how it's done [here](https://github.com/ruma/ruma/blob/56f9f48a6a8a727e38847057a2ae112f4b1008e3/crates/ruma-client-api/src/membership/join_room_by_id_or_alias.rs#L200) in Ruma.